appmonitoring/ts/src/CertificateGenerator.ts

```diff
@@ -232,6 +233,53 @@ export class CertificateManager {
         return caCertificate.verify(certificate);
     }
 
+    private static IsValidCertificate(operationId: string, mwhcCaBundle: string, webhookCertData: WebhookCertData, clusterArmId: string, clusterArmRegion: string): boolean {
+        try {
+            forge.pki.certificateFromPem(mwhcCaBundle);
+            forge.pki.certificateFromPem(webhookCertData.caCert);
+            forge.pki.certificateFromPem(webhookCertData.tlsCert);
+            forge.pki.privateKeyFromPem(webhookCertData.tlsKey);
+            return true;
+        } catch (error) {
+            logger.error('Error occured while trying to validate certificates!', operationId, this.requestMetadata);
+            logger.error(JSON.stringify(error), operationId, this.requestMetadata);
+            logger.SendEvent("CertificateValidationFailed", operationId, null, clusterArmId, clusterArmRegion, true, error);
+            return false;
+        }
+    }
+
+    private static async CheckCertificateJobStatus(kubeConfig: k8s.KubeConfig, operationId: string, clusterArmId: string, clusterArmRegion: string): Promise<boolean> {
+        kubeConfig.loadFromDefault();
+
+        const k8sApi = kubeConfig.makeApiClient(k8s.BatchV1Api);
+        const requestMetadata = this.requestMetadata;
+
+        async function checkJobStatus(namespace, jobName) : Promise<boolean> {
```

Comment:
Why do we need a local function that is only invoked once immediately below?

Reply:
Let's not do it please unless there's an actual reason, it's weird and arbitrary. If you feel you need a separate method - just make it a normal method. If not - just have the content of the method directly in the body of the top level method.

Reply:
wait thats not the line I thought you were talking about